kong/plugins/cors/handler.lua

```diff
@@ -7,12 +7,24 @@ CorsHandler.PRIORITY = 2000
 
 local OPTIONS = "OPTIONS"
 
+local function host_in_domain(domain, conf)
+  for i, d in ipairs(conf.origin) do
+    if string.match(domain, '[%.|//]'..d..'$') then
```

Comment:
Let's do a couple things here:

- use the PCRE API offered by `ngx.re`: https://github.com/openresty/lua-nginx-module#ngxrematch
- build the regex ahead of time instead of on every request